src/libraries/System.Net.Http/src/System/Net/Http/Headers/HeaderDescriptor.cs

```diff
@@ -15,32 +16,69 @@ namespace System.Net.Http.Headers
     // Use HeaderDescriptor.TryGet to resolve an arbitrary header name to a HeaderDescriptor.
     internal readonly struct HeaderDescriptor : IEquatable<HeaderDescriptor>
     {
-        private readonly string _headerName;
-        private readonly KnownHeader? _knownHeader;
+        /// <summary>
+        /// Either a <see cref="KnownHeader"/> or <see cref="string"/>
+        /// </summary>
+        private readonly object? _descriptor;
```

Comment:
It won't be null unless you observe a `default` instance of it. Since we do intentionally use an uninitialized `HeaderDescriptor` in a few places, I made the field nullable (to force null checks / asserts on properties/methods).

Reply:
Ok, can we add a comment explaining this?